dwave/preprocessing/src/presolveimpl.hpp

```diff
@@ -53,9 +53,8 @@ class PresolverImpl {
     explicit PresolverImpl(model_type model) : model_(std::move(model)) {}
 
     /// Apply any loaded presolve techniques. Acts on the model() in-place.
-    void apply() {
-        normalize();
-        presolve();
+    bool apply() {
+        return normalize() | presolve();
```

Comment:
Might make sense to leave a comment here noting that you are intentionally *not* short-circuiting

Reply:
Alternatively, use the pattern we use below...

Reply:
Added a comment